### PR TITLE
Add top level imports

### LINKED ISSUE #113: Top level imports
Including imports at the top level of the package can be beneficial from the user's point of view, similar to what cirq has done in its [top-level](https://github.com/quantumlib/Cirq/blob/main/cirq-core/cirq/__init__.py).

What are your thoughts on this?

### PR DESCRIPTION
**Pull request by inmzhang**
Fixes #113

I'm not sure whether we need the `__all__` statements in each `__init__.py`. I understand it served as a convention for public api, but it's kind of cumbersome to add every method name twice within a single `__init__.py`.

### CODE REVIEW

File: src/tqec/templates/atomic/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from .square import (
+    AlternatingSquareTemplate,
+    AlternatingCornerSquareTemplate,
+)
+
+__all__ = [
+    "AlternatingRectangleTemplate",
+    "RawRectangleTemplate",
+    "AlternatingSquareTemplate",
+    "AlternatingCornerSquareTemplate",
```
Comment (nelimee) [inline src/tqec/templates/atomic/__init__.py:15]: The `AlternatingCornerSquareTemplate` is currently on hold. If I remember correctly, the implementation uses permutations and I found at least one issue with that approach, as noted in the docstring:
https://github.com/QCHackers/tqec/blob/c1bf3d44697b77f04be9ee36436dca606d07b505/src/tqec/templates/atomic/square.py#L60-L63
I should have opened an issue back then, it seems like I forgot, I will do that just after finishing this review.
In the meantime, I am in favour of commenting out this class from the top-level imports, as a weak sign that it is not yet ready for the public API.

File: src/tqec/plaquette/__init__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    RoundedPlaquette,
+    PlaquetteList,
+)
+
+from .qubit import PlaquetteQubit
+
+from .schedule import (
+    ScheduleException,
+    ScheduledCircuit,
+    ScheduledCircuits,
```
Comment (nelimee) [inline src/tqec/plaquette/__init__.py:13]: The `ScheduledCircuits` class is only an implementation detail and does not appear in the public API, I am in favour of removing it from the top-level imports.
```suggestion
```

File: src/tqec/templates/atomic/__init__.py (hunk 1 above)
Comment (nelimee) [reply src/tqec/templates/atomic/__init__.py:15]:
```suggestion
```

File: src/tqec/templates/atomic/__init__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from .rectangle import (
+    AlternatingRectangleTemplate,
+    RawRectangleTemplate,
+)
+
+from .square import (
+    AlternatingSquareTemplate,
+    AlternatingCornerSquareTemplate,
```
Comment (nelimee) [inline src/tqec/templates/atomic/__init__.py:8]:
```suggestion
```

File: src/tqec/plaquette/__init__.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+__all__ = [
+    "Plaquette",
+    "SquarePlaquette",
+    "RoundedPlaquette",
+    "PlaquetteList",
+    "PlaquetteQubit",
+    "ScheduleException",
+    "ScheduledCircuit",
+    "ScheduledCircuits",
```
Comment (nelimee) [inline src/tqec/plaquette/__init__.py:24]:
```suggestion
```

File: src/tqec/detectors/__init__.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from .operation import (
+    ShiftCoords,
+    RelativeMeasurementData,
+    RelativeMeasurementsRecord,
+    Detector,
+    Observable,
```
Comment (nelimee) [inline src/tqec/detectors/__init__.py:6]: Do we want to expose `ShiftCoords`, `Detector` and `Observable` here? Not exposing them in the top-level API would be a good sign to tell to the user "if you have to dig into the library to get something, you should know what you are doing, so read the documentation first" in my opinion.

Comment (nelimee) [reply src/tqec/detectors/__init__.py:6]: Also, I am not sure that the `RelativeMeasurementsRecord` should be exposed. Does it have another usage than avoiding code duplication between `Observable` and `Detector`?

**Review by nelimee — COMMENTED** [review]
I have made per-file comments. 

I also have a question: do we need to import everything on the top-level `tqec` module? I think a user would be fine with `from tqec.templates import AlternatingSquareTemplate` (adding the `.template`). It would reduce the number of entries in the top level module and would, in my opinion, make it easier to discover for new users.

As a side note, the `__all__` attribute in the `__init__.py` file was initially added by me to silence quickly a linter warning about unused imports. I find that quite annoying now that I see it on several files with more imports, and see no point in doing that. I am in favour of removing that.

**Comment by inmzhang** [comment]
> also have a question: do we need to import everything on the top-level tqec module? 

Yes, we can encapulate some modules within the sub-packages and not expose them directly in the top-level. I think the imports hierachy can be better determined when #134 is addressed.

**Review by nelimee — APPROVED** [review]
LGTM

**Pull request merged**